### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**
[DEMO LINK](https://Zelinskyi-Serhii.github.io/react_movies-list-add-form/)

### CODE REVIEW

**Review by milenadiamonds1 — APPROVED** [review]
GJ

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const handleAddMovie = (movie: Movie) => {
+    const all = [...movies, movie];
```
Comment (sergii-nosachenko) [inline src/App.tsx:12]: Use more specific name

File: src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+    setMovies(all);
+  };
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
```
Comment (sergii-nosachenko) [inline src/App.tsx:22]:
```suggestion

      <div className="sidebar">
```

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,96 @@
...
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isActiveButton = (
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:19]: The button is not active with this condition. Use proper naming.

File: src/components/NewMovie/NewMovie.tsx [hunk 4]
```diff
@@ -1,45 +1,96 @@
...
+    title.length === 0
+    || imgUrl.length === 0
+    || imdbUrl.length === 0
+    || imdbId.length === 0);
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:23]: Check if any value is empty too (only spaces)

Screenshot 2023-04-04 at 12 11 33

File: src/components/NewMovie/NewMovie.tsx [hunk 5]
```diff
@@ -1,45 +1,96 @@
...
+
+  const reset = () => {
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:25]: Use more specific name

File: src/components/NewMovie/NewMovie.tsx [hunk 6]
```diff
@@ -1,45 +1,96 @@
...
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={() => {
+        setCount(count + 1);
+        addMovie();
+      }}
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:53]: Create handler for this

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 7]
```diff
@@ -8,7 +8,7 @@ import { Movie } from './types/Movie';
 export const App = () => {
   const [movies, setMovies] = useState(moviesFromServer);
 
-  const handleAddMovie = (movie: Movie) => {
+  const addNewMovie = (movie: Movie) => {
     const all = [...movies, movie];
```
Comment (sergii-nosachenko) [inline src/App.tsx]:
```suggestion
  const addNewMovie = (newMovie: Movie) => {
    const allMovies = [...movies, newMovie];
```

File: src/components/NewMovie/NewMovie.tsx [hunk 8]
```diff
@@ -16,21 +16,21 @@ export const NewMovie: React.FC<Props> = ({ onAdd }) => {
...
-  const isActiveButton = (
-    title.length === 0
-    || imgUrl.length === 0
-    || imdbUrl.length === 0
-    || imdbId.length === 0);
+  const isNotActiveButton = (
+    title.trim().length === 0
+    || imgUrl.trim().length === 0
+    || imdbUrl.trim().length === 0
+    || imdbId.trim().length === 0);
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:23]:
```suggestion
  const isFilled = (value: string): boolean => value.trim().length;
  const isActiveAddButton = [
    title,
    imgUrl,
    imdbUrl,
    imdbId,
  ].every(isFilled);
```

File: src/components/NewMovie/NewMovie.tsx [hunk 9]
```diff
@@ -99,7 +102,7 @@ export const NewMovie: React.FC<Props> = ({ onAdd }) => {
             type="submit"
             data-cy="submit-button"
             className="button is-link"
-            disabled={isActiveButton}
+            disabled={isNotActiveButton}
```
Comment (sergii-nosachenko) [inline src/components/NewMovie/NewMovie.tsx:105]:
```suggestion
            disabled={!isActiveAddButton}
```

**Review by sergii-nosachenko — APPROVED** [review]
GJ.
Commented on how you can improve your code.

**Pull request closed without merging**